collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -1040,8 +1066,8 @@ private boolean saveAnswersForCurrentScreen(boolean evaluateConstraints) {
         // only try to save if the current event is a question or a field-list group
         // and current view is an ODKView (occasionally we show blank views that do not have any
         // controls to save data from)
-        if (formController.currentPromptIsQuestion() && currentView instanceof ODKView) {
-            HashMap<FormIndex, IAnswerData> answers = ((ODKView) currentView)
+        if (formController.currentPromptIsQuestion() && getCurrentViewIfODKView() != null) {
+            LinkedHashMap<FormIndex, IAnswerData> answers = getCurrentViewIfODKView()
```

Comment:
We should always refer to a Collection using its Interface and not its Implementation (in case we want to swap out implementations later).

Reply:
Oh yes, good catch!